lib/3scale_toolbox.rb

```diff
@@ -1,9 +1,32 @@
+require '3scale_toolbox/configuration'
+
 module ThreeScaleToolbox
+  @config_file = nil
```

Comment:
Is this really needed to be module level? 

I imagine it is possible to provide an instance of `ThreeScaleToolbox::Configuration` to the `--config-file` option. Then there would be no need to "remember" the file that was passed in parameters and modify global state (good luck in tests). 

And every command would have it available as an option.

Reply:
Would be nice to "inject" configuration object as part of options available to all commands. Do not know how, though.

Config file has setter method and can be injected. Then calling configuration object for the first time, that config file will be used.

Reply:
After reading cri source and docs I'm not impressed. There there is no builtin support transformations of options/arguments or validations. 
The only thing I found is `option_added` delegate to do some transformations. IMO we could use that to transform the configuration file name into the configuration object, that would be passed to every command in options. Then every command would get own instance of this object based on the CLI flag / env / default value.
But the "delegate" feature looks very internal and not really usable from the outside.